### PR TITLE
Feature/button enable hold

### PR DESCRIPTION
**Pull request by vlach1989**

### CODE REVIEW

File: src/Button/index.js [hunk 1]
```diff
@@ -28,21 +30,95 @@ class Button extends React.PureComponent {
...
+
+	onTouchEnd(evt) {
+		evt.stopPropagation();
+		evt.preventDefault();
+		this.onMouseOut(evt);
+	}
+
+	onMouseDown() {
+		if (this.props.holdEnabled && !this.props.disabled) {
+			this.props.onClick();
```
[the remark this answers is not in the document]
Comment (vlach1989) [reply src/Button/index.js]: Fixed while reviewing :) Check previous commit

File: src/Button/index.js [hunk 2]
```diff
@@ -14,11 +14,13 @@ class Button extends React.PureComponent {
 		className: PropTypes.string,
 		disabled: PropTypes.bool,
 		ghost: PropTypes.bool,
+		holdEnabled: PropTypes.bool,
```
Comment (tmdmc) [inline src/Button/index.js:17]: Just test if the handler is present

File: src/Button/index.js [hunk 3]
```diff
@@ -28,21 +30,94 @@ class Button extends React.PureComponent {
...
 		this.onKeyPress = this.onKeyPress.bind(this);
+
+		this.onMouseOut = this.onMouseOut.bind(this);
+		this.onMouseDown = this.onMouseDown.bind(this);
+		this.onTouchStart = this.onTouchStart.bind(this);
+		this.onTouchEnd = this.onTouchEnd.bind(this);
+	}
+
+	componentDidMount() {
+		this.node.current.addEventListener('touchstart', this.onTouchStart);
```
Comment (tmdmc) [inline src/Button/index.js:59]: Maybe listeners should be added when we know we have an onHold handler?

File: src/Button/index.js [hunk 4]
```diff
@@ -28,21 +30,94 @@ class Button extends React.PureComponent {
...
 	}
 
+	componentDidUpdate(prevProps, prevState, snapshot) {
+		if (this.props.disabled && this.state.held) {
+			this.clearHoldTimeout();
+		}
+	}
+
+	onTouchStart(evt) {
+		evt.stopPropagation();
```
Comment (tmdmc) [inline src/Button/index.js:75]: Is it ok to stop propagation and/or preventDefault, when we're not actually doing anything ourselves?
E.g. if there's no onHold handler, don't we want long click to be a click?

File: src/Button/index.js [hunk 5]
```diff
@@ -28,21 +30,94 @@ class Button extends React.PureComponent {
 	};
 
 	static defaultProps = {
-		disabled: false
+		disabled: false,
+		onHoldCallStart: 300,
+		onHoldCallStep: 50,
```
Comment (tmdmc) [inline src/Button/index.js:35]:
```suggestion
		holdStart: 300,
		holdStep: 50,
```
Don't use 'on*' when it's not an event handler (even if it's related)

File: src/Button/index.js (hunk 4 above)
Comment (tmdmc) [reply src/Button/index.js:75]: (maybe one or both of those should be inside the if, or maybe if we add listeners when we have a handler, it'll be a moot point)

**Pull request merged**